app/code/Magento/Elasticsearch/Model/DataProvider/Base/Suggestions.php

```diff
@@ -100,7 +107,8 @@ public function __construct(
         ConnectionManager $connectionManager,
         SearchIndexNameResolver $searchIndexNameResolver,
         StoreManager $storeManager,
-        FieldProviderInterface $fieldProvider,
+        CollectionFactory $productAttributeCollectionFactory,
```

Comment:
To maintain backwards/upstream compatibility, the old argument needs to stay, and the new arguments need to go after $getSuggestionFrequency (line 113) as optional parameters. Then you need to use ObjectManager to get the instance if it isn't passed in, like core does with `logger` and `getSuggestionFrequency` (lines 124-126).

This way, any code that uses or extends Suggestions.php can still work even with the old constructor arguments, and the new code here will fill in anything missing. No BC break.